modules/gis/src/lib/geojson-to-binary.js

```diff
@@ -73,87 +68,133 @@ function firstPass(features) {
 
           // eslint-disable-next-line max-depth
           for (const coord of flatten(polygon)) {
-            // eslint-disable-next-line max-depth
-            if (coord.length === 3) maxCoordLength = 3;
+            coordLengths.add(coord.length);
           }
         }
         break;
       default:
         throw new Error(`Unsupported geometry type: ${geometry.type}`);
     }
+
+    if (feature.properties) {
+      for (const key in feature.properties) {
+        const val = feature.properties[key];
+        numericProps[key] = numericProps[key] ? isNumeric(val) : numericProps[key];
+      }
+    }
   }
 
   return {
     pointPositions,
     linePositions,
     linePaths,
-    coordLength: maxCoordLength,
+    coordLength: Math.max(...coordLengths),
     polygonPositions,
     polygonObjects,
-    polygonRings
+    polygonRings,
+    // Array of keys whose values are always numeric
+    numericProps: Object.keys(numericProps).filter(k => numericProps[k])
   };
 }
 
 // Second scan over GeoJSON features
 // Fills coordinates into pre-allocated typed arrays
-function secondPass(features, firstPassData, options = {}) {
+function secondPass(features, options = {}) {
   const {
     pointPositions,
     linePositions,
     linePaths,
     coordLength,
     polygonPositions,
     polygonObjects,
-    polygonRings
-  } = firstPassData;
-  const {PositionDataType = Float32Array} = options;
+    polygonRings,
+    numericProps,
+    PositionDataType = Float32Array
+  } = options;
   const points = {
     positions: new PositionDataType(pointPositions * coordLength),
-    objectIds: new Uint32Array(pointPositions)
+    globalFeatureIndex: new Uint32Array(pointPositions),
```

Comment:
I know we decided to do this later, but if we could add a count of the features:

`globalFeatureIndex: globalFeatureCount > 65535 ? new Uint32Array(pointPositions) : new Uint16Array(pointPositions),`

Reply:
I updated this to use `Uint16Array` where possible both for the feature indices and for the `pathIndices`, `polygonIndices`, `primitivePolygonIndices`. (Aside, now that I think about it more, `polygonRingIndices` seems like a more apt name, since the term "linear ring" is formalized in the GeoJSON spec).

I also updated to count features by geometry type, so if there are 66,000 features, but split equally as 22,000 of point/line/polygon, the `globalFeatureIndex` arrays would be `Uint32`, but the local ones would be `Uint16`

Reply:
Awesome! Just consider what test coverage you think we need for this, and since you are attending to a lot of little details, I'd recommend making sure documentation is updated as you go  through each of them 

(This is a general comment, I haven't checked if / what you added in tests/docs in the last commits).